src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1630,6 +1654,61 @@ private void MakeSeparatorList(ReadOnlySpan<char> separators, ref ValueListBuild
             }
         }
 
+        private void MakeSeparatorListVectorized(ref ValueListBuilder<int> sepListBuilder, char c, char? c2 = null, char? c3 = null)
+        {
+            Vector256<byte> shuffleConstant = Vector256.Create(0x02, 0x06, 0x0A, 0x0E, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+                                                        0xFF, 0xFF, 0xFF, 0xFF, 0x02, 0x06, 0x0A, 0x0E, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF);
+
+            Vector256<ushort> v1 = Vector256.Create(c);
+            Vector256<ushort>? v2 = c2 is char sep2 ? Vector256.Create(sep2) : (Vector256<ushort>?)null;
+            Vector256<ushort>? v3 = c3 is char sep3 ? Vector256.Create(sep3) : (Vector256<ushort>?)null;
+
+            ref char c0 = ref MemoryMarshal.GetReference(this.AsSpan());
+            int cond = Length - (Length % Vector256<ushort>.Count);
+            int i = 0;
+
+            for (; i < cond; i += Vector256<ushort>.Count)
+            {
+                ref char ci = ref Unsafe.Add(ref c0, i);
+                Vector256<ushort> charVector = Unsafe.As<char, Vector256<ushort>>(ref ci);
```

Comment:
Should be `ReadUnlaligned` instead of `As`?

Reply:
I honestly have no idea, this is what I've seen when I was looking at instrinsics examples in C#. Maybe somebody else can comment on this?

Reply:
We don't know whether the data is aligned or not.
Current JIT does an unaligned read for `As`, but it's not documented and could change in the future. Hence `ReadUnaligned` is the safer option. 
Although with current JIT codegen should be the same.

Reply:
Ah I see, makes sense! `ReadUnaligned` does not seem to take a ref char however, how would I convert from `ref char` to `ref byte`?

Reply:
With `Unsafe.As` :wink:

For this I usually create a small helper method (or local function), so that the code remains clean:
```c#
static Vector256<ushort> ReadVector(ref char src)
{
    ref byte b = ref Unsafe.As<char, byte>(ref src);
    return Unsafe.ReadUnaligned<Vector256<byte>>(ref b);
}
```
Endiannes is no concern here, as AVX2 implies little-endian.

Reply:
Ah I see, makes sense! I think I tried it earlier, but must have forgotten a `ref` before the function call! Vielen Dank fuer die schnelle Hilfe :)

Reply:
Fixed in https://github.com/dotnet/runtime/pull/38001/commits/52ad5acd1fe5dc4ecc830828b68077d70e05e757

Reply:
Just to note: a different approach would be to pin the string, then use `Avx.LoadVector` to read from the pointer.
With this way, it's easy to eliminate / avoid the bound-check in the processing of the sequential remainder.